### PR TITLE
fix: quotes participants validation

### PR DESCRIPTION
**Pull request by vijayg10**

### CODE REVIEW

**Comment by kleyow** [comment]
Do these fixes fix https://github.com/mojaloop/project/issues/3951?

**Comment by vijayg10** [comment]
> Do these fixes fix [mojaloop/project#3951](https://github.com/mojaloop/project/issues/3951)?

No @kleyow I think that's a different issue and can be handled in that story.

File: src/model/quotes.js [hunk 1]
```diff
@@ -170,11 +170,14 @@ class QuotesModel {
...
     // In fspiop api spec 2.0, to support FX, `supportedCurrencies` can be optionally passed in via the payer property.
     // If `supportedCurrencies` is present, then payer FSP must have position accounts for all those currencies.
-    // If it is not passed in, then we validate against the `amount` currency only.
-    const payerFspCurrencies = quoteRequest.payer.supportedCurrencies || [quoteRequest.amount.currency]
-    await Promise.all(payerFspCurrencies.map(async currency => {
-      await this.db.getParticipant(fspiopSource, LOCAL_ENUM.PAYER_DFSP, currency, ENUM.Accounts.LedgerAccountType.POSITION)
-    }))
+    if (quoteRequest.payer.supportedCurrencies && quoteRequest.payer.supportedCurrencies.length > 0) {
+      await Promise.all(quoteRequest.payer.supportedCurrencies.map(async currency => {
+        await this.db.getParticipant(fspiopSource, LOCAL_ENUM.PAYER_DFSP, currency, ENUM.Accounts.LedgerAccountType.POSITION)
```
Comment (geka-evk) [inline src/model/quotes.js:175]: Why do we need await iside `.map()`?
```
await Promise.all(quoteRequest.payer.supportedCurrencies.map(
  currency => this.db.getParticipant(fspiopSource, LOCAL_ENUM.PAYER_DFSP, currency, ENUM.Accounts.LedgerAccountType.POSITION)
)  
```

**Review by geka-evk — APPROVED** [review]

File: src/model/quotes.js (hunk 1 above)
Comment (vijayg10) [reply src/model/quotes.js]: You are right, refactored. Thanks

**Review by oderayi — APPROVED** [review]

**Pull request merged**